tests/test_client.py

```diff
@@ -194,6 +194,385 @@ def test_search_camera_all_correct_Format_Error(self, mock_get):
+        clientId = '0' * 96
+        clientSecret = '0' * 71
+        client = Client(clientId, clientSecret)
+        # provide token for building header
+        client.token = "correctToken"
+        # manipulate request.post's result
+        mock_response = mock.Mock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {
+            "clientID": "test_clientID",
+            "clientSecret": "test_clientSecret"
+        }
+        mock_post.return_value = mock_response
+        # validate result
+        expected_clientID = 'test_clientID'
+        expected_clientSecret = 'test_clientSecret'
+        url = Client.base_URL + 'apps/register'
+        header = {'Authorization': 'Bearer correctToken'}
+        data = {'owner': 'testowner', 'permissionLevel': 'user'}
+        clientID, clientSecret = client.register('testowner')
+        mock_post.assert_called_once_with(url, headers=header, data=data)
+        self.assertEqual(clientID, expected_clientID)
+        self.assertEqual(clientSecret, expected_clientSecret)
+        self.assertEqual(2, mock_response.json.call_count)
+
+    @mock.patch('pythonAPIClient.client.requests.post')
+    def test_register_incorrect_token(self, mock_post):
+        clientId = '0' * 96
+        clientSecret = '0' * 71
+        client = Client(clientId, clientSecret)
+        # manipulate request.post's result
+        mock_response = mock.Mock()
+        mock_response.status_code = 401
+        mock_response.json.return_value = {
+            "message": "Failed to authenticate token"
+        }
+        mock_post.return_value = mock_response
+        # validate result
+        url = Client.base_URL + 'apps/register'
+        data = {'owner': 'testowner', 'permissionLevel': 'user'}
+        header = {'Authorization': 'Bearer None'}
+        with self.assertRaises(AuthenticationError):
+            client.register('testowner')
+        mock_post.assert_called_once_with(url, headers=header, data=data)
+        self.assertEqual(2, mock_response.json.call_count)
+
+    @mock.patch('pythonAPIClient.client.requests.post')
+    @mock.patch('pythonAPIClient.client.requests.get')
+    def test_register_expired_token(self, mock_get, mock_post):
+        clientId = '0' * 96
+        clientSecret = '0' * 71
+        client = Client(clientId, clientSecret)
+        client.token = 'ExpiredToken'
+        # set request.post's result
+        mock_response = mock.Mock()
+        mock_response.status_code = 401
+        mock_response.json.return_value = {
+            "message": "Token expired"
+        }
+        mock_post.return_value = mock_response
+        with self.assertRaises(InternalError):
+            client.register('testowner')
+        url = Client.base_URL + 'apps/register'
+        data = {'owner': 'testowner', 'permissionLevel': 'user'}
+        header = {'Authorization': 'Bearer ExpiredToken'}
+        mock_post.assert_called_once_with(url, headers=header, data=data)
```

Comment:
how does this function work? if the token expires, shouldn't the request get sent twice?